### ISSUE TITLE
SA1306 should also check fields starting with a single underscore

### ISSUE THREAD

**Issue by vweijsters** [DotNetAnalyzers/StyleCopAnalyzers#1861]
The following code should trigger SA1306 in my opinion (with SA1309 disabled and SX1309 enabled).

```csharp
public class TestClass
{
    private int _TestField;
}
```

I would propose that we make the underscore for fields a property and that we adapt the different analyzers and code fixes to take this setting into account properly. I think with the correct set of properties we can wrap up SA1309, SX1309 and SX1309S into a single analyzer.

**Comment by sharwell** [comment]
I'd like to wait and see how dotnet/roslyn#4616 and dotnet/roslyn#7064 are handled before I go too far on this. What do you think?

**Comment by vweijsters** [comment]
I'd prefer to solve this on a relatively short time frame, as I normally have SX1309 enabled and would prefer to also have SA1306.

**Comment by sharwell** [comment]
I'd be OK with simply ignoring underscores for SA1306, regardless of the state of SA1309 and/or SX1309.
